src/worker/tasks/chainIndexer.ts

```diff
@@ -0,0 +1,356 @@
+        decodedEventName = parsedLog.name;
+        decodedLog = parsedLog.eventFragment.inputs.reduce((acc, input) => {
+          acc[input.name] = {
+            type: input.type,
+            value: parsedLog.args[input.name].toString(),
+          };
+          return acc;
+        }, {} as Record<string, { type: string; value: string }>);
+      } catch (error) {
+        logger({
+          service: "worker",
+          level: "warn",
+          message: `Failed to decode log: chainId: ${params.chainId}, contractAddress ${contractAddress}`,
+        });
+      }
+    }
+
+    const block = blockCache[log.blockNumber];
+
+    // format the log entry
+    return {
+      chainId: params.chainId,
+      blockNumber: log.blockNumber,
+      contractAddress: log.address.toLowerCase(), // ensure common address handling across
+      transactionHash: log.transactionHash,
+      topic0: log.topics[0],
+      topic1: log.topics[1],
+      topic2: log.topics[2],
+      topic3: log.topics[3],
+      data: log.data,
+      eventName: decodedEventName,
+      decodedLog: decodedLog,
+      timestamp: new Date(block.timestamp * 1000), // ethers timestamp is s, Date uses ms
+      transactionIndex: log.transactionIndex,
+      logIndex: log.logIndex,
+    };
+  });
+
+  return formattedLogs;
+};
+
+interface IndexFnParams {
+  pgtx: PrismaTransaction;
+  chainId: number;
+  fromBlockNumber: number;
+  toBlockNumber: number;
+  subscribedContractAddresses: string[];
+}
+
+const indexContractEvents = async ({
+  pgtx,
+  chainId,
+  fromBlockNumber,
+  toBlockNumber,
+  subscribedContractAddresses,
+}: IndexFnParams) => {
+  // get all logs for the contracts
+  const logs = await getSubscribedContractsLogs({
+    chainId,
+    fromBlock: fromBlockNumber,
+    toBlock: toBlockNumber,
+    contractAddresses: subscribedContractAddresses,
+  });
+
+  // update the logs
+  if (logs.length > 0) {
+    await bulkInsertContractEventLogs({ logs, pgtx });
+  }
+};
+
+const indexTransactionReceipts = async ({
+  pgtx,
+  chainId,
+  fromBlockNumber,
+  toBlockNumber,
+  subscribedContractAddresses,
+}: IndexFnParams) => {
+  const { blocks, transactionsWithReceipt } = await getBlocksAndTransactions({
+    chainId,
+    fromBlock: fromBlockNumber,
+    toBlock: toBlockNumber,
+    contractAddresses: subscribedContractAddresses,
+  });
+
+  const blockLookup = blocks.reduce((acc, curr) => {
+    acc[curr.number] = curr;
+    return acc;
+  }, {} as Record<number, BlockWithTransactions>);
+
+  const txReceipts = transactionsWithReceipt.map(({ receipt, transaction }) => {
+    return {
+      chainId: chainId,
+      blockNumber: receipt.blockNumber,
+      contractAddress: receipt.to.toLowerCase(),
+      contractId: getContractId(chainId, receipt.to.toLowerCase()),
+      transactionHash: receipt.transactionHash.toLowerCase(),
+      blockHash: receipt.blockHash.toLowerCase(),
+      timestamp: new Date(blockLookup[receipt.blockNumber].timestamp),
```

Comment:
this needs to be updated.

Looks like all dates are coming incorrect, `1970-01-20 19:06:37.024`. 

We need to multiply by `1000` as the timestamp is in Unix Epoc (seconds